src/Agent/NewRelic/Agent/Core/WireModels/LoadedModuleWireModelCollection.cs

```diff
@@ -12,9 +16,206 @@ public class LoadedModuleWireModelCollection
     {
         public List<LoadedModuleWireModel> LoadedModules { get; }
 
-        public LoadedModuleWireModelCollection()
+        private LoadedModuleWireModelCollection()
         {
             LoadedModules = new List<LoadedModuleWireModel>();
         }
+
+        public static LoadedModuleWireModelCollection Build(IList<Assembly> assemblies)
```

Comment:
If you have time to do some performance testing, I'd be curious how long this `Build` step takes. Calculating hashes is generally quite expensive. I have one pretty easy optimization suggestion below.

Reply:
I have not done any perf testing on this since it only happens once per assembly at startup (already a packed and slow spot, I know).  Looking at suggestion!